src/aggregation/Aggregator.ts

```diff
@@ -243,3 +244,116 @@ export class LongSumAggregator extends AbstractAggregator<Long> {
+
+    readData(input: DataInput): any {
+        this.attributePath = input.readString();
+        const count = input.readInt();
+        for (let i = 0; i < count; i++) {
+            input.readObject();
+        }
+    }
+
+    writeData(output: DataOutput): void {
+        output.writeString(this.attributePath);
+        output.writeInt(0);
+    }
+}
+
+/** @internal */
+export class CanonicalizingHashSet<R> implements IdentifiedDataSerializable, Set<R> {
+
+    classId = AggregatorConstants.CANONICALIZING_SET;
+    factoryId = AggregatorConstants.AGGREGATOR_FACTORY_ID;
+    private _values = new Set<R>();
+
+    readData(input: DataInput): void {
+        const count = input.readInt();
+        for (let i = 0; i < count; i++) {
+            const element = input.readObject();
+            this._values.add(element);
+        }
+    }
+
+    writeData(output: DataOutput): void {
+        output.writeInt(this._values.size);
+        for (const element of this._values) {
+            output.writeObject(element);
+        }
+    }
+
+    readonly [Symbol.toStringTag]: string = this._values[Symbol.toStringTag];
+
+    get size() : number {
+        return this._values.size;
+    }
+
+    [Symbol.iterator](): IterableIterator<R> {
+        return undefined;
+    }
+
+    add(value: R): this {
+        this._values.add(value);
+        return this;
+    }
+
+    clear(): void {
+        this._values.clear();
+    }
+
+    delete(value: R): boolean {
+        return this._values.delete(value);
+    }
+
+    entries(): IterableIterator<[R, R]> {
+        return this._values.entries();
+    }
+
+    forEach(callbackfn: (value: R, value2: R, set: Set<R>) => void, thisArg?: any): void {
+        this._values.forEach(callbackfn, thisArg);
+    }
+
+    has(value: R): boolean {
+        return this._values.has(value);
+    }
+
+    keys(): IterableIterator<R> {
+        return this._values.keys();
+    }
+
+    values(): IterableIterator<R> {
+        return this._values.values();
+    }
+
+}
+
+/** @internal */
+export const idToConstructor: { [id: number]: new () => Aggregator<any> } = {
+    [AggregatorConstants.COUNT]: CountAggregator,
+    [AggregatorConstants.DISTINCT]: DistinctValuesAggregator,
+    [AggregatorConstants.DOUBLE_AVG]: DoubleAverageAggregator,
+    [AggregatorConstants.DOUBLE_SUM]: DoubleSumAggregator,
+    [AggregatorConstants.FIXED_SUM]: FixedPointSumAggregator,
+    [AggregatorConstants.FLOATING_POINT_SUM]: FloatingPointSumAggregator,
+    [AggregatorConstants.INT_AVG]: IntegerAverageAggregator,
+    [AggregatorConstants.INT_SUM]: IntegerSumAggregator,
+    [AggregatorConstants.LONG_AVG]: LongAverageAggregator,
+    [AggregatorConstants.LONG_SUM]: LongSumAggregator,
+    [AggregatorConstants.MAX]: MaxAggregator,
+    [AggregatorConstants.MIN]: MinAggregator,
+    [AggregatorConstants.NUMBER_AVG]: NumberAverageAggregator,
+    [AggregatorConstants.CANONICALIZING_SET]: CanonicalizingHashSet
+};
```

Comment:
I had to move this here, otherwise `idToContructor` was having undefined as values instead of classes